app/js/containers/Artifacts.jsx

```diff
@@ -30,27 +42,28 @@ const ArtifactsContainer = ({ artifacts, onArtifactClick }) => {
                             </tr>
                         </thead>
                         <tbody>
-                            { Object.keys(artifacts).map(id =>
+                            { artifacts.map((artifact, index) =>
                                 <Artifact
-                                    key={id}
-                                    data={artifacts[id]}
-                                    onClick={() => onArtifactClick(id)}
+                                    key={index}
```

Comment:
I suspect using `index` as a `key` isn't a good idea. As the store changes, entirely unrelated artifacts will share the same key, resulting in `react` assuming it doesn't need to redraw those components.